File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/WhatIfPagerAdapter.java
package xyz.jienan.xkcd.whatif;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import xyz.jienan.xkcd.home.base.BaseStatePagerAdapter;
import xyz.jienan.xkcd.whatif.fragment.SingleWhatIfFragment;

public class WhatIfPagerAdapter extends BaseStatePagerAdapter {

    public WhatIfPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        SingleWhatIfFragment fragment = SingleWhatIfFragment.Companion.newInstance(position + 1);
        fragmentsMap.put(position + 1, fragment);
        return fragment;
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/WhatIfPagerAdapter.kt
package xyz.jienan.xkcd.whatif

import androidx.fragment.app.Fragment
import xyz.jienan.xkcd.home.base.BaseStatePagerAdapter
import xyz.jienan.xkcd.whatif.fragment.SingleWhatIfFragment

class WhatIfPagerAdapter(fragment: Fragment) : BaseStatePagerAdapter(fragment) {

    override fun createFragment(position: Int) = SingleWhatIfFragment.newInstance(position + 1)
}